src/transfer.h

```diff
@@ -67,25 +78,23 @@ void transfer_to_function(
   // Vector of expansion_coefficients to be set
   Eigen::Matrix<T, Eigen::Dynamic, 1>& expansion_coefficients = f->x()->array();
 
-  int idx = 0;
+  int row_offset = 0;
   for (int c = 0; c < ncells; ++c)
   {
-    const int np = cell_particles[c].size();
-    Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> q(
-        space_dimension, np * value_size);
-    Eigen::VectorXd l(np * value_size);
-    for (int p = 0; p < np; ++p)
-    {
-      int pidx = cell_particles[c][p];
-      Eigen::Map<const Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic,
-                                     Eigen::RowMajor>>
-          basis(basis_values.row(idx++).data(), space_dimension, value_size);
+    std::vector<int> cell_particles = pax.cell_particles()[c];
+    std::pair<Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>,
```

Comment:
C++17 allows `auto [q, l]` here which is much neater...

Reply:
Didn't know, nice usage of `auto`! Code has been updated in https://github.com/LEoPart-X/LEoPart/pull/16/commits/0d492c9b7c91fbb825a0c85c5e2a2cfb2329282b

Reply:
Yes. Also `std::tie(q, l)` can be used if already allocated.